### PR TITLE
Expose fs_base/gs_base registers.

### PR DESCRIPTION
**Pull request by aarzilli**
I'm trying to use rr as a backend for [delve](http://github.com/derekparker/delve) a debugger for [go](http://golang.org) that's used by [several go IDE's](https://github.com/derekparker/delve/blob/master/Documentation/EditorIntegration.md).

The only problem I have encountered is that go uses the FS segment to store a pointer to a structure describing the currently running goroutine, so delve needs to read the value of fs_base but AFAICT there isn't a way to ask rr about it. 

With this change I added fs_base and gs_base to target.xml, it's based [on a patch to do the same thing to gdbserver](https://www.sourceware.org/ml/gdb-patches/2015-11/msg00076.html) that AFAIK never got finished. 

The test set passes and gdb doesn't seem to mind.

### CODE REVIEW

File: src/Registers.cc [hunk 1]
```diff
@@ -169,14 +169,11 @@ RegisterInfo<rr::X64Arch>::Table RegisterInfo<rr::X64Arch>::registers = {
   // The comparison for this is handled specially
   // elsewhere.
   RV_X64_WITH_MASK(ORIG_RAX, orig_rax, 0),
+  RV_X64_WITH_MASK(FS_BASE, fs_base, 8),
+  RV_X64_WITH_MASK(GS_BASE, gs_base, 8),
```
Comment (rocallahan) [inline src/Registers.cc]: I think these can just use `RV_X64`.

File: src/Registers.cc [hunk 2]
```diff
@@ -91,7 +91,7 @@ template <> struct RegisterInfo<rr::X64Arch> {
   static const size_t num_registers = DREG_NUM_LINUX_X86_64;
   typedef RegisterTable<num_registers> Table;
   static Table registers;
-  static RegisterValue non_gdb_registers[2];
+  static RegisterValue non_gdb_registers[0];
```
Comment (rocallahan) [inline src/Registers.cc]: With this patch there are no longer any `non_gdb_registers` so you should just remove all related code.

File: src/Registers.cc (hunk 1 above)
Comment (rocallahan) [inline src/Registers.cc]: To clarify: a mask of 8 doesn't make much sense. Using `RV_X64` means we'll die if `fs_base`/`gs_base` get out of sync during replay, but that's probably a good thing!

**Comment by aarzilli** [comment]
Done.

**Comment by rocallahan** [comment]
Thanks!

**Comment by aarzilli** [comment]
Thank you!

**Pull request merged**